cpp/tests/utilities/column_wrapper.cuh

```diff
@@ -0,0 +1,396 @@
+      }
+
+      // If one value is valid but the other is not
+      if (lhs_is_valid xor rhs_is_valid) {
+        return false;
+      }
+
+      return nulls_are_equivalent;
+    }
+  };
+
+  /**---------------------------------------------------------------------------*
+   * @brief Compares if another column_wrapper is equal to this wrapper.
+   *
+   * Treats NULL == NULL
+   *
+   * @param rhs  The other column_wrapper to check for equality
+   * @return true The two columns are equal
+   * @return false The two columns are not equal
+   *---------------------------------------------------------------------------**/
+  bool operator==(column_wrapper<ColumnType> const& rhs) const {
+    if (the_column.size != rhs.the_column.size) return false;
+    if (the_column.dtype != rhs.the_column.dtype) return false;
+    if (the_column.null_count != rhs.the_column.null_count) return false;
+    if (the_column.dtype_info.time_unit != rhs.the_column.dtype_info.time_unit)
+      return false;
+
+    if (!(the_column.data && rhs.the_column.data))
+      return false;  // if one is null but not both
+
+    if (not thrust::all_of(rmm::exec_policy()->on(0),
+                           thrust::make_counting_iterator(0),
+                           thrust::make_counting_iterator(the_column.size),
+                           elements_equal{the_column, rhs.the_column})) {
+      return false;
+    }
+
+    CUDA_RT_CALL(cudaPeekAtLastError());
+
+    return true;
+  }
+
+ private:
+  /**---------------------------------------------------------------------------*
+   * @brief Allocates and initializes the underyling gdf_column with host data.
+   *
+   * Creates a gdf_column and copies data from the host for it's data and
+   * bitmask. Sets the corresponding dtype based on the column_wrapper's
+   * ColumnType.
+   *
+   * @param host_data The vector of host data to copy to device for the column's
+   * data
+   * @param host_bitmask Optional vector of host data for the column's bitmask
+   * to copy to device
+   *---------------------------------------------------------------------------**/
+  void initialize_with_host_data(
+      std::vector<ColumnType> const& host_data,
+      std::vector<gdf_valid_type> const& host_bitmask =
+          std::vector<gdf_valid_type>{}) {
+    // thrust::device_vector takes care of host to device copy assignment
+    data = host_data;
+
+    // Fill the gdf_column members
+    the_column.data = data.data().get();
+    the_column.size = data.size();
+    the_column.dtype = cudf::gdf_dtype_of<ColumnType>();
+    gdf_dtype_extra_info extra_info;
+    extra_info.time_unit = TIME_UNIT_NONE;
+    the_column.dtype_info = extra_info;
+
+    // If a validity bitmask vector was passed in, allocate device storage
+    // and copy its contents from the host vector
+    if (host_bitmask.size() > 0) {
+      gdf_size_type const required_bitmask_size{
+          gdf_get_num_chars_bitmask(host_data.size())};
```

Comment:
The function named `gdf_get_num_chars_bitmask()` has _got_ to be renamed. I'm sure I renamed it at least a couple of times... :-(

Reply:
Absolutely. This is why this code exists: https://github.com/rapidsai/cudf/blob/branch-0.6/cpp/src/bitmask/bit_mask.h#L63

It's a TODO to update the code base to _use_ this new functionality. In the meantime, people should continue to use the old and clunky `gdf_get_num_chars_bitmask` so that it's easier to do find & replace.